src/app/components/search/DateRangeFilter.js

```diff
@@ -68,127 +88,194 @@ function parseEndDateAsISOString(moment) {
   return new Date(date.getFullYear(), date.getMonth(), date.getDate(), 23, 59, 59).toISOString();
 }
 
-class DateRangeFilter extends React.Component {
-  get valueType() {
-    const { value } = this.props;
+const DateRangeFilter = ({
+  classes,
+  hide,
+  value,
+  onChange,
+  onRemove,
+}) => {
+  const getValueType = () => {
     if (value && value.updated_at) return 'updated_at';
-    if (value && value.last_seen) return 'last_seen';
-    if (value && value.published_at) return 'published_at';
+    if (value && value.media_published_at) return 'media_published_at';
+    if (value && value.report_published_at) return 'report_published_at';
     return 'created_at';
-  }
+  };
+  const rangeTypes = {
+    startEnd: 'startEnd',
+    relative: 'relative',
+    none: '',
+  };
+  const getInitialRangeType = () => {
+    if (value && value[getValueType()]?.condition) return rangeTypes.relative;
+    if (value && value[getValueType()]?.start_time) return rangeTypes.startEnd;
+    return rangeTypes.none;
+  };
+  const [rangeType, setRangeType] = React.useState(getInitialRangeType());
+  const relativeRanges = {
```

Comment:
I used this kind of "struct of constants" pattern to provide easy remapping in case the API changes

Reply:
@dariusk I can change back-end to accept `days`, `weeks`, `months` or `years` instead of `d`, `w`, `m` or `y` in case it's cleaner and easier in front-end what do you think?

Reply:
@melsawy I think it's okay as it currently is. It's good practice on the front end to be able to map variables like this anyhow. APIs change, and so do frontend models for what things mean. Or maybe we want to change all our frontend variables to be, who knows, Hungarian notation in the future. This just lets us define the mapping between the frontend and the API however we like, and keep all the "magic strings" in one place.